Jetcaster/app/src/main/java/com/example/jetcaster/ui/home/Home.kt

```diff
@@ -142,6 +154,69 @@ private fun <T> ThreePaneScaffoldNavigator<T>.isMainPaneHidden(): Boolean {
     return scaffoldValue[SupportingPaneScaffoldRole.Main] == PaneAdaptedValue.Hidden
 }
 
+@OptIn(ExperimentalMaterial3AdaptiveApi::class)
+fun calculateScaffoldDirective(
+    windowAdaptiveInfo: WindowAdaptiveInfo,
+    verticalHingePolicy: HingePolicy = HingePolicy.AvoidSeparating
+): PaneScaffoldDirective {
+    val maxHorizontalPartitions: Int
+    val verticalSpacerSize: Dp
+    if (windowAdaptiveInfo.windowSizeClass.isCompact()) {
+        maxHorizontalPartitions = 1
+        verticalSpacerSize = 0.dp
+    } else {
+        when (windowAdaptiveInfo.windowSizeClass.windowWidthSizeClass) {
+            WindowWidthSizeClass.COMPACT -> {
+                maxHorizontalPartitions = 1
+                verticalSpacerSize = 0.dp
+            }
+            WindowWidthSizeClass.MEDIUM -> {
+                maxHorizontalPartitions = 1
+                verticalSpacerSize = 0.dp
+            }
+            else -> {
+                maxHorizontalPartitions = 2
+                verticalSpacerSize = 24.dp
+            }
+        }
+    }
+    val maxVerticalPartitions: Int
+    val horizontalSpacerSize: Dp
+
+    if (windowAdaptiveInfo.windowPosture.isTabletop) {
+        maxVerticalPartitions = 2
+        horizontalSpacerSize = 24.dp
+    } else {
+        maxVerticalPartitions = 1
+        horizontalSpacerSize = 0.dp
+    }
+
+    val defaultPanePreferredWidth = 360.dp
+
+    return PaneScaffoldDirective(
+        maxHorizontalPartitions,
+        verticalSpacerSize,
+        maxVerticalPartitions,
+        horizontalSpacerSize,
+        defaultPanePreferredWidth,
+        getExcludedVerticalBounds(windowAdaptiveInfo.windowPosture, verticalHingePolicy)
+    )
+}
+
+@OptIn(ExperimentalMaterial3AdaptiveApi::class)
+private fun getExcludedVerticalBounds(posture: Posture, hingePolicy: HingePolicy): List<Rect> {
+    return when (hingePolicy) {
+        HingePolicy.AvoidSeparating -> posture.separatingVerticalHingeBounds
+        HingePolicy.AvoidOccluding -> posture.occludingVerticalHingeBounds
+        HingePolicy.AlwaysAvoid -> posture.allVerticalHingeBounds
+        else -> emptyList()
+    }
+}
+
+private fun androidx.window.core.layout.WindowSizeClass.isCompact(): Boolean =
```

Comment:
There are currently two different `WindowSizeClass` symbols, one in `androidx-window` (used by the `adaptive` deps), and one in `material3-window-size-class`. I think things are moving towards the `andriodx-window` one, and I have a mind to migrate all usages of the other, but not sure if this PR is the right place for that.

Reply:
Thanks for calling that out. Yeah we can make that migration on a separate PR